downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1563 @@
+
+public class Listener {
+   ...
+   ...
+   // listener life cycle methods
+
+   public function init() {
+      ...
+      // object init failed due to some reason
+      if (object_init_failed) {
+         panic error("Listener initialization failed");
+      }
+   }
+}
+```
+
+##### Referring Lang Library Modules Without Using Quoted Identifiers
+
+Lang library module prefixes can now be used without the initial quote. For example, both the approaches below are now supported.
+
+```ballerina
+int a = int:sum(1, 2);
+```
+
+```ballerina
+int a = 'int:sum(1, 2);
+```
+
+The way of overriding a langlib module is also relaxed now. For example, both the approaches below are now supported.
+
+```ballerina
+import somePackage.someModule as string;
+```
+
+```ballerina
+import somePackage.someModule as `string;
+```
+
+##### Improved Lang Library Methods
+
+###### Introduction of the `includes` Method
+
+A new lang library method named `includes`, which tests whether a `string` value includes another `string` value has been added to the `ballerina/lang.string` library. It accepts an optional second argument to indicate the index to start searching from. It returns `true` only if the string contains the other string at an index greater or equal to the start index. The value `false` will be returned otherwise.
+
+```ballerina
+string str = "Ballerina Programming Language";
+boolean contains = str.includes("Language", 10);
+```
+
+###### Introduction of the `sleep` Method
+
+A new `sleep` method, which pauses the execution of the current strand for a specified time in seconds has been added to the `ballerina/lang.runtime` library.
+
+```ballerina
+runtime:sleep(2.0);
+```
+
+###### Introduction of the `getStackTrace` Method
+
+A new `getStackTrace` method has been introduced in the `ballerina/lang.runtime` library to get a stack trace for the current call stack for the specified execution point. It returns an array of stack frames. 
+
+```ballerina
+runtime:StackFrame[] stackFrames = runtime:getStackTrace();
+```
+
+###### Introduction of the `Cloneable` Type
+
+A new `Cloneable` type has been introduced to the `ballerina/lang.value` library. This is a cyclic union type. This type represents the values on which `clone` and `cloneReadOnly` can be applied on.
+
+```ballerina
+public type Cloneable readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>;
+```
+
+#### Rename of the `ballerina/java` Module
+
+The `ballerina/java` module is renamed to `ballerina/jballerina.java`.
+
+#### Runtime
+
+New APIs are added for creating immutable Ballerina records and arrays from Java. 
+
+The new `ValueCreator.createReadonlyArrayValue` API creates a new `readonly` Ballerina array from the given Java array and the `ValueCreator.createReadonlyRecordValue` creates a new `readonly` Ballerina record.
+
+##### Configurable
+
+The `configurable` feature is improved to support variables with decimal and arrays of int, float, string, boolean and decimal types.
```

Comment:
```suggestion
The `configurable` feature is improved to support variables with decimal and arrays of `int`, `float`, `string`, `boolean`, and `decimal` types.
```

Reply:
@HindujaB, do we need a simple example here of how the value can be specified in the toml file?

Reply:
We already have TOML example for the previously supported types in the slp7 release notes. 
We can add as following though, 
```suggestion
The `configurable` feature is improved to support variables with decimal and arrays of `int`, `float`, `string`, `boolean`, and `decimal` types.

The format of the `Config.toml` file is as follows.

```toml
[pkgName.submoduleName]
# The above toml table specification is not needed for single file execution
intVar = 24
floatVar = 4.8
stringVar = "hello world"
booleanVar = false
decimalVar = 2.33

intArray = [1,2,3]
floatArray = [9.0, 5.6]
stringArray = ["red", "yellow", "green"]
booleanArray = [true, false, false, true]
decimalArray = [8.9, 4.5, 6.2]
```
```

Reply:
Sorry, what I meant was for the list. 

What if we add something like

````
The arrays needs to be specified in the TOML file as follows.

```toml
[pkgName.submoduleName]
intArray = [1,2,3]
stringArray = ["red", "yellow", "green"]
```
````

Unless we can assume the user would know this. Looping in @pubudu91 too.

Reply:
+1

Reply:
I think it's ok to assume prior knowledge about TOML since it's a standard format. What we need to show in samples is the format of a config TOML file, not the TOML syntax. And maybe how it would then be accessed in the code

Reply:
How about something like

````md
For example if the `Config.toml` file contains the following arrays 

```toml
ints = [1,2,3]
strings = ["red", "yellow", "green"]
```

they can be loaded as follows.

```ballerina
configurable int[] & readonly ints = ?;
configurable string[] & readonly strings = ?;
```
````

Reply:
This sample looks good to me.

Reply:
Updated.

Reply:
````suggestion

For example if the `Config.toml` file contains the following TOML arrays, 

```toml
ints = [1,2,3]
strings = ["red", "yellow", "green"]
```
they can be loaded with following configurable variables.

```ballerina
configurable int[] & readonly ints = ?;
configurable string[] & readonly strings = ?;
```

````